x-pack/legacy/plugins/reporting/server/lib/create_queue.ts

```diff
@@ -5,29 +5,19 @@
  */
 
 import { ElasticsearchServiceSetup } from 'kibana/server';
-import {
-  ServerFacade,
-  ExportTypesRegistry,
-  HeadlessChromiumDriverFactory,
-  QueueConfig,
-  Logger,
-} from '../../types';
+import { ESQueueInstance, ServerFacade, QueueConfig, Logger } from '../../types';
+import { ReportingPlugin } from '../plugin';
 // @ts-ignore
 import { Esqueue } from './esqueue';
 import { createWorkerFactory } from './create_worker';
 import { createTaggedLogger } from './create_tagged_logger'; // TODO remove createTaggedLogger once esqueue is removed
 
-interface CreateQueueFactoryOpts {
-  exportTypesRegistry: ExportTypesRegistry;
-  browserDriverFactory: HeadlessChromiumDriverFactory;
-}
-
-export function createQueueFactory(
+export async function createQueueFactory<JobParamsType, JobPayloadType>(
+  reporting: ReportingPlugin,
   server: ServerFacade,
   elasticsearch: ElasticsearchServiceSetup,
-  logger: Logger,
-  { exportTypesRegistry, browserDriverFactory }: CreateQueueFactoryOpts
-): Esqueue {
```

Comment:
Since more accessors exist on the `reporting` object, I have simplified signatures like this because the extra `opts` no longer need to be passed separately